unfoldingWord/translationCore
# redux resources

#### This pull request addresses:

This PR aims to eliminate the usage of the following api calls:
```
api.getDataFromCommon('originalLanguage'); **greek/hebrew**
api.getDataFromCommon('gatewayLanguage'); **ULB/UDB**
api.getDataFromCommon('targetLanguage');
api.getDataFromCheckStore('TranslationHelps', 'wordList');
api.getDataFromCheckStore('TranslationHelps', 'sectionList')
```
Now all of these resources will be pass down as props from resourcesReducer

**TODO: still need to load this resources to the resourcesReducer with the new project load implementation. therefore resourcesReducer only has all resources when fetch data in run.**

#### How to test this pull request:

1. To test this branch you must load a fresh project.
2. pull all the branches mentioned below.
3. log this.props in app.js render method and make sure you see the resourcesReducer with bibles 
(originalLanguage, targetLanguage, ULB and UDB) and resources (tN or tW)

This PR is related to the following PRs and branches in the tools repos:

- https://github.com/translationCoreApps/translationWords_Check_plugin/pull/49 (feature/mc-fetchdata-redux)

- https://github.com/translationCoreApps/translationNotes_Check_plugin/pull/44 (feature/mc-fetchdata-redux)

- https://github.com/translationCoreApps/ScripturePane/pull/22 (feature/mc-fetchdata-redux)


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/unfoldingword-dev/translationcore/924)
